### ISSUE TITLE
[Bug] MergedOptions UpdateMergedOptionsFromMicrosoftIdentityOptions error

### ISSUE THREAD

**Issue by JadynWong** [AzureAD/microsoft-identity-web#1215]
**Which version of Microsoft Identity Web are you using?**
Note that to get help, you need to run the latest version. 
1.11.0
1.10.0

**Where is the issue?**
* Web app
    * [x] Sign-in users
    * [ ] Sign-in users and call web APIs
* Web API
    * [ ] Protected web APIs (validating tokens)
    * [ ] Protected web APIs (validating scopes)
    * [ ] Protected web APIs call downstream web APIs
* Token cache serialization
     * [x] In-memory caches
     * [ ] Session caches
     * [ ] Distributed caches
* Other (please describe)

**Is this a new or an existing app?**
The app is in development and I have upgraded to a new version of Microsoft Identity Web.

**Repro**

```csharp
context.Services.AddAuthentication()
                .AddMicrosoftIdentityWebApp(
                    configuration: configuration,
                    configSectionName: "AzureAd",
                    openIdConnectScheme: OpenIdConnectDefaults.AuthenticationScheme,
                    cookieScheme: null,
                    subscribeToOpenIdConnectMiddlewareDiagnosticsEvents: false
                );
```

```json
  "AzureAd": {
    "Instance": "https://login.microsoftonline.com/",
    "Domain": "contoso.com",
    "TenantId": "xxxxxxxxxxxx-xxxx-xxxx-xxxxxxxxxxxx",
    "ClientId": "xxxxxxxxxxxx-xxxx-xxxx-xxxxxxxxxxxx",
    "ClientSecret": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
    "CallbackPath": "/signin-oidc",
    "SignedOutCallbackPath ": "/signout-callback-oidc",
    "ResponseType": "code id_token",
    "Scope": [ "email" ]
  },
```

https://github.com/AzureAD/microsoft-identity-web/blob/222e76a60930949e3e4838ece6e5ef49e793f091/src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs#L275-L291

https://github.com/AzureAD/microsoft-identity-web/blob/222e76a60930949e3e4838ece6e5ef49e793f091/src/Microsoft.Identity.Web/MergedOptions.cs#L184-L192

**Expected behavior**
ResponseType should is  `code id_token`
Scope should is `openid profile email`

**Actual behavior**
ResponseType should is  `id_token`
Scope should is `openid profile`

**Possible solution**

```c#
 Configure<OpenIdConnectOptions>(OpenIdConnectDefaults.AuthenticationScheme, options =>
{
    options.Scope.Add("email");
    options.ResponseType = OpenIdConnectResponseType.CodeIdToken;
});
```
Some properties of OpenIdConnectOptions have default values, such as ResponseType,Scope,ResponseMode etc.
There are some problems with the current MergedOptions method.

**Additional context / logs / screenshots**
![code](https://user-images.githubusercontent.com/13699353/119607058-47f1dd80-be26-11eb-9ddc-042fa5fbd112.gif)

**Comment by jmprieur** [comment]
Thanks @JadynWong for the heads-up and for your analysis

**Comment by jmprieur** [comment]
@JadynWong: this is fixed in 1.12.0
